### ISSUE TITLE
Missing papers from ICLR2023

### ISSUE THREAD

**Issue by NanLi2021** [pranftw/openreview_scraper#1]
Hi, I tried your package on ICLR2023 with the keyword "disentangling" but the paper "Continual Unsupervised Disentangling of Self-Organizing Representations" is not returned, which seems wrong. The paper is clearly accepted [https://openreview.net/forum?id=ih0uFRFhaZZ](https://openreview.net/forum?id=ih0uFRFhaZZ). 

Could you please look into this issue? Thanks!

**Comment by pranftw** [comment]
Hey @NanLi2021, thanks for the issue. The problem was that I used a function that checked for keywords in keywords instead of keywords in text, so that caused it to not search properly within the title and abstract. I've fixed it in the latest commit(b850276). I've checked that the paper you were referring to is now returned properly.

Code
```python
from scraper import Scraper
from extract import Extractor
from filters import title_filter, keywords_filter, abstract_filter
from utils import save_papers, load_papers


years = [
    '2023'
]
conferences = [
    'ICLR'
]
keywords = [
    'disentangling'
]

def modify_paper(paper):
  paper.forum = f"https://openreview.net/forum?id={paper.forum}"
  paper.content['pdf'] = f"https://openreview.net{paper.content['pdf']}"
  return paper

extractor = Extractor(fields=['forum'], subfields={'content':['title', 'keywords', 'abstract', 'pdf', 'match']})
scraper = Scraper(conferences=conferences, years=years, keywords=keywords, extractor=extractor, fpath='example.csv', fns=[modify_paper])

scraper.add_filter(title_filter)
scraper.add_filter(keywords_filter)
scraper.add_filter(abstract_filter)

scraper()
```

CSV
```
forum,title,keywords,abstract,pdf,match,venue,year,type
https://openreview.net/forum?id=d5U-bPKPde,Group-Disentangling Conditional Shift,"['group disentanglement', 'variational autoencoders', 'conditional shift']","We propose a novel group disentanglement method called the Context-Aware Variational Autoencoder (CxVAE). Our model can learn disentangled representations on datasets with conditional shift. This phenomenon occurs when the conditional distribution of the instance-level latent variable $\mathbf{z}$ given the input observation $\mathbf{x}$ changes from one group to another (i.e. $p_i(\mathbf{z}|\mathbf{x}) \neq p_j(\mathbf{z}|\mathbf{x})$, where $i,j$ are two different groups). We show that existing methods fail to learn disentangled representations under this scenario because they infer the group $\mathbf{u}$ and instance $\mathbf{z}$ variables separately. CxVAE overcomes this limitation by conditioning the instance inference on the group variable $q(\mathbf{z}|\mathbf{x},\mathbf{u})$. Our model has the novel ability to disentangle ambiguous observations (those with incomplete information about the generative factors), which we evaluate on the task of fair comparisons between student test scores. Additionally, we demonstrate empirically that conditional shift is the cause of our model's improved performance.",https://openreview.net/pdf/b934fcad1f7445dafa78fb6f29acc619f0cd0e51.pdf,{'title_filter': 'disentangling'},ICLR.cc,2023,Conference
https://openreview.net/forum?id=9Z_GfhZnGH,Disentanglement with Biological Constraints: A Theory of Functional Cell Types,"['Disentangling', 'neurosciece', 'representation learning', 'hippocampus', 'cortex']","Neurons in the brain are often finely tuned for specific task variables. Moreover, such disentangled representations are highly sought after in machine learning. Here we mathematically prove that simple biological constraints on neurons, namely nonnegativity and energy efficiency in both activity and weights, promote such sought after disentangled representations by enforcing neurons to become selective for single factors of task variation. We demonstrate these constraints lead to disentanglement in a variety of tasks and architectures, including variational autoencoders. We also use this theory to explain why the brain partitions its cells into distinct cell types such as grid and object-vector cells, and also explain when the brain instead entangles representations in response to entangled task factors. Overall, this work provides a mathematical understanding of why single neurons in the brain often represent single human-interpretable factors, and steps towards an understanding task structure shapes the structure of brain representation.",https://openreview.net/pdf/868eda4d383fc687b3d363b3f173ce871a57f087.pdf,{'keywords_filter': 'disentangling'},ICLR.cc,2023,Conference
https://openreview.net/forum?id=NeDc-Ak-H_,Learning What and Where: Disentangling Location and Identity Tracking Without Supervision,"['object permanence', 'CATER', 'unsupervised learning', 'binding problem']","Our brain can almost effortlessly decompose visual data streams into background and salient objects. Moreover, it can anticipate object motion and interactions, which are crucial abilities for conceptual planning and reasoning. Recent object reasoning datasets, such as CATER, have revealed fundamental shortcomings of current vision-based AI systems, particularly when targeting explicit object representations, object permanence, and object reasoning. Here we introduce a self-supervised LOCation and Identity tracking system (Loci), which excels on the CATER tracking challenge. Inspired by the dorsal and ventral pathways in the brain, Loci tackles the binding problem by processing separate, slot-wise encodings of 'what' and 'where'. Loci's predictive coding-like processing encourages active error minimization, such that individual slots tend to encode individual objects. Interactions between objects and object dynamics are processed in the disentangled latent space. Truncated backpropagation through time combined with forward eligibility accumulation significantly speeds up learning and improves memory efficiency. Besides exhibiting superior performance in current benchmarks, Loci effectively extracts objects from video streams and separates them into location and Gestalt components. We believe that this separation offers a representation that will facilitate effective planning and reasoning on conceptual levels.",https://openreview.net/pdf/6b48752e87eb85e35fe0dc79498e9bf30671acde.pdf,{'title_filter': 'disentangling'},ICLR.cc,2023,Conference
https://openreview.net/forum?id=CW6KmU5wPh,DAVA: Disentangling Adversarial Variational Autoencoder,"['Disentanglement learning', 'varational auto-encoder', 'curriculum learning', 'generative adversarial networks']","The use of well-disentangled representations poses many advantages for downstream tasks, e.g. increasing sample efficiency, or enabling interpretability. Their quality is, however, determined to a large extent by the choice of dataset-specific hyperparameters, most notably the regularization strength. To address the issue, we introduce DAVA, a novel training procedure for variational auto-encoders that alleviates the issue of hyperparameter selection at the cost of a comparatively small overhead. We compare DAVA against models with optimal choice of hyperparameters. Without any hyperparameter tuning, DAVA is competitive across a diverse range of commonly used datasets.
Further, even under an adequate set of hyperparameters, the success of the disentanglement process remains heavily influenced by randomness in network initialization. We therefore present the new unsupervised PIPE disentanglement metric, capable of evaluating representation quality. We demonstrate the PIPE metrics ability to positively predict performance of downstream models in abstract reasoning. We also exhaustively examine correlations with existing supervised and unsupervised metrics.",https://openreview.net/pdf/f5547864776700331caee15a950d9cf20a918029.pdf,{'title_filter': 'disentangling'},ICLR.cc,2023,Conference
https://openreview.net/forum?id=LE5LxBgjB4V,Disentangling the Mechanisms Behind Implicit Regularization in SGD,"['deep learning', 'generalization', 'implicit regularization', 'sgd']","A number of competing hypotheses have been proposed to explain why small-batch Stochastic Gradient Descent (SGD) leads to improved generalization over the full-batch regime, with recent work crediting the implicit regularization of various quantities throughout training. However, to date, empirical evidence assessing the explanatory power of these hypotheses is lacking. In this paper, we conduct an extensive empirical evaluation, focusing on the ability of various theorized mechanisms to close the small-to-large batch generalization gap. Additionally, we characterize how the quantities that SGD has been claimed to (implicitly) regularize change over the course of training. By using micro-batches, i.e. disjoint smaller subsets of each mini-batch,  we empirically show that explicitly penalizing the gradient norm or the Fisher Information Matrix trace, averaged over micro-batches, in the large-batch regime recovers small-batch SGD generalization, whereas Jacobian-based regularizations fail to do so. This generalization performance is shown to often be correlated with how well the regularized model's gradient norms resemble those of small-batch SGD. We additionally show that this behavior breaks down as the micro-batch size approaches the batch size. Finally, we note that in this line of inquiry, positive experimental findings on CIFAR10 are often reversed on other datasets like CIFAR100, highlighting the need to test hypotheses on a wider collection of datasets.",https://openreview.net/pdf/dd9347e40eef7d9c26eccf2185107afa06e7dab1.pdf,{'title_filter': 'disentangling'},ICLR.cc,2023,Conference
https://openreview.net/forum?id=EMvG1Jdhw_8,Disentangling Learning Representations with Density Estimation,"['autoencoder', 'representation learning', 'disentanglement', 'density estimation']","Disentangled learning representations have promising utility in many applications, but they currently suffer from serious reliability issues. We present Gaussian Channel Autoencoder (GCAE), a method which achieves reliable disentanglement via scalable non-parametric density estimation of the latent space. GCAE avoids the curse of dimensionality of density estimation by disentangling subsets of its latent space with the Dual Total Correlation (DTC) metric, thereby representing its high-dimensional latent joint distribution as a collection of many low-dimensional conditional distributions. In our experiments, GCAE achieves highly competitive and reliable disentanglement scores compared with state-of-the-art baselines.",https://openreview.net/pdf/6b884b796156f663352f3b95ed1bc2d91dac4202.pdf,{'title_filter': 'disentangling'},ICLR.cc,2023,Conference
https://openreview.net/forum?id=Ojpb1y8jflw,"StyleMorph: Disentangling Shape, Pose and Appearance through 3D Morphable Image and Geometry Generation ","['3D-aware GAN', 'Template-based', 'Morphable', 'Disentanglement', 'Photorealistic', 'Neural Radiance Field', 'StyleGAN']","We introduce StyleMorph, a 3D generative model that relies on the 3D morphable model paradigm to disentangle shape, pose, object and scene texture for high quality image synthesis. We represent 3D shape variability through 3D deformation fields with respect to a canonical object template. Both the deformations and the template are expressed as implicit networks and learned in an unsupervised manner only from 2D image supervision. We connect 3D morphable modelling with deferred neural rendering by performing an implicit surface rendering of “Template Object Coordinates” (TOCS), thereby constructing a purely geometric, deformation-equivariant 2D signal that reflects the compounded geometric effects of non-rigid shape, pose, and perspective projection. We use TOCS maps in tandem with object and background appearance codes to condition a StyleGAN-based deferred neural rendering (DNR) network for high-resolution image synthesis. We show competitive photorrealistic image synthesis results on 4 datasets (FFHQ faces, AFHQ Cats, Dogs, Wild), while achieving the joint disentanglement of shape, pose, object and scene texture. ",https://openreview.net/pdf/f389290b72106a796ac49d64f9d23622f9f2b76f.pdf,{'title_filter': 'disentangling'},ICLR.cc,2023,Conference
https://openreview.net/forum?id=TFbwV6I0VLg,SlotFormer: Unsupervised Visual Dynamics Simulation with Object-Centric Models,"['Object-centric learning', 'dynamics modeling', 'Transformer']","Understanding dynamics from visual observations is a challenging problem that requires disentangling individual objects from the scene and learning their interactions. While recent object-centric models can successfully decompose a scene into objects, modeling their dynamics effectively still remains a challenge. We address this problem by introducing SlotFormer -- a Transformer-based autoregressive model operating on learned object-centric representations. Given a video clip, our approach reasons over object features to model spatio-temporal relationships and predicts accurate future object states. In this paper, we successfully apply SlotFormer to perform video prediction on datasets with complex object interactions. Moreover, the unsupervised SlotFormer's dynamics model can be used to improve the performance on supervised downstream tasks, such as Visual Question Answering (VQA), and goal-conditioned planning. Compared to past works on dynamics modeling, our method achieves significantly better long-term synthesis of object dynamics, while retaining high quality visual generation. Besides, SlotFormer enables VQA models to reason about the future without object-level labels, even outperforming counterparts that use ground-truth annotations. Finally, we show its ability to serve as a world model for model-based planning, which is competitive with methods designed specifically for such tasks.",https://openreview.net/pdf/1316e463c439d163c66066ccca66c1587da678bb.pdf,{'abstract_filter': 'disentangling'},ICLR.cc,2023,Conference
https://openreview.net/forum?id=ih0uFRFhaZZ,Continual Unsupervised Disentangling of Self-Organizing Representations,"['continual disentanglment', 'generative model', 'VAE', 'SOM']","Limited progress has been made in continual unsupervised learning of representations, especially in reusing, expanding, and continually disentangling learned semantic factors across data environments. We argue that this is because existing approaches treat continually-arrived data independently, without considering how they are related based on the underlying semantic factors. We address this by a new generative model describing a topologically-connected mixture of spike-and-slab distributions in the latent space, learned end-to-end in a continual fashion via principled variational inference. The learned mixture is able to automatically discover the active semantic factors underlying each data environment and to accumulate their relational structure based on that. This distilled knowledge of different data environments can further be used for generative replay and guiding continual disentangling of new semantic factors. We tested the presented method on a split version of 3DShapes to provide the first quantitative disentanglement evaluation of continually learned representations, and further demonstrated its ability to continually disentangle new representations in benchmark datasets.",https://openreview.net/pdf/d207be46a521d6688563744319d7c9f7446fcce9.pdf,{'title_filter': 'disentangling'},ICLR.cc,2023,Conference
https://openreview.net/forum?id=LI4mXhTg23M,Learning Disentanglement in Autoencoders through Euler Encoding,"['disentanglement', 'disentangling', 'linear disentangled representations', 'autoencoder', 'latent space', 'factorizing', 'latent-space factorization', 'latent-space regularization']","Noting the importance of factorizing (or disentangling) the latent space, we propose a novel, non-probabilistic disentangling framework for autoencoders, based on the principles of symmetry transformations that are independent of one another. To the best of our knowledge, this is the first deterministic model that is aiming to achieve disentanglement based on autoencoders without pairs of images or labels, by explicitly introducing inductive biases into a model architecture through Euler encoding. The proposed model is then compared with a number of state-of-the-art models, relevant to disentanglement, including symmetry-based and generative models based on autoencoders. Our evaluation using six different disentanglement metrics, including the unsupervised disentanglement metric we propose here in this paper, shows that the proposed model can offer better disentanglement, especially when variances of the features are different, where other methods may struggle. We believe that this model opens several opportunities for linear disentangled representation learning based on deterministic autoencoders.",https://openreview.net/pdf/5acdfd43e7450df4fe2e43bef364c1b3ca51ae2a.pdf,{'keywords_filter': 'disentangling'},ICLR.cc,2023,Conference
```
I'll close this issue for now. Please feel free to reach out if you need anymore help!

Thanks for using this repo!
Cheers!

**Comment by NanLi2021** [comment]
Yes, it works now. Thank you so much!
